cloudformation/ec2.yaml

```diff
@@ -0,0 +1,335 @@
+  the AWS resources used if you create a stack from this template.
+Parameters:
+  KeyName:
+    Description: Name of an existing EC2 KeyPair to enable SSH access to the instance
+    Type: AWS::EC2::KeyPair::KeyName
+    ConstraintDescription: Must be the name of an existing EC2 KeyPair.
+  VpcId:
+    Description: >-
+      ID of the VPC within which the EC2 instance would be setup. You can find instructions
+      for setup at https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Scenarios.html
+      Example https://docs.aws.amazon.com/vpc/latest/userguide/vpc-subnets-commands-example.html
+    Type: AWS::EC2::VPC::Id
+    ConstraintDescription: Must be a valid VPC id.
+  SubnetId:
+    Description: >-
+      ID of the Subnet within which the EC2 instance would be setup.
+      Check VpcId Parameter Description for setup instructions
+    Type: AWS::EC2::Subnet::Id
+    ConstraintDescription: Must be a valid VPC id.
+  InstanceType:
+    Description: TorchServe EC2 instance type
+    Type: String
+    Default: c4.4xlarge
+    AllowedValues:
+      - t1.micro
+      - t2.nano
+      - t2.micro
+      - t2.small
+      - t2.medium
+      - t2.large
+      - m1.small
+      - m1.medium
+      - m1.large
+      - m1.xlarge
+      - m2.xlarge
+      - m2.2xlarge
+      - m2.4xlarge
+      - m3.medium
+      - m3.large
+      - m3.xlarge
+      - m3.2xlarge
+      - m4.large
+      - m4.xlarge
+      - m4.2xlarge
+      - m4.4xlarge
+      - m4.10xlarge
+      - c1.medium
+      - c1.xlarge
+      - c3.large
+      - c3.xlarge
+      - c3.2xlarge
+      - c3.4xlarge
+      - c3.8xlarge
+      - c4.large
+      - c4.xlarge
+      - c4.2xlarge
+      - c4.4xlarge
+      - c4.8xlarge
+      - g2.2xlarge
+      - g2.8xlarge
+      - p2.xlarge
+      - p2.8xlarge
+      - p2.16xlarge
+      - p3.2xlarge
+      - p3.8xlarge
+      - p3.16xlarge
+      - r3.large
+      - r3.xlarge
+      - r3.2xlarge
+      - r3.4xlarge
+      - r3.8xlarge
+      - i2.xlarge
+      - i2.2xlarge
+      - i2.4xlarge
+      - i2.8xlarge
+      - d2.xlarge
+      - d2.2xlarge
+      - d2.4xlarge
+      - d2.8xlarge
+    ConstraintDescription: Must be a valid EC2 instance type.
+  ManagementLocation:
+    Description: 'The IP address range that can call management endpoint of TorchServe'
+    Type: String
+    MinLength: '9'
+    MaxLength: '18'
+    Default: '0.0.0.0/0'
+    AllowedPattern: (\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})/(\d{1,2})
+    ConstraintDescription: Must be a valid IP CIDR range of the form x.x.x.x/x.
+  SSHLocation:
+    Description: 'The IP address range that can be used to SSH to the EC2 instances'
+    Type: String
+    MinLength: '9'
+    MaxLength: '18'
+    Default: '0.0.0.0/0'
+    AllowedPattern: (\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})/(\d{1,2})
+    ConstraintDescription: Must be a valid IP CIDR range of the form x.x.x.x/x.
+Mappings:
+  AWSRegion2AMI:
+    us-east-1:
+      AMI: ami-085925f297f89fce1
```

Comment:
@maaquib Are these the DLAMIs in each of the regions or a new AMI for TorchServe? Again we have maintenance overhead by explicitly calling out. Using the parameter config option in CFN as described in will simplify things: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/retrieve-ecs-optimized_AMI.html. This can be done as an optimization step in next release.

Reply:
@chauhang Just confirming if there is a requirement to use DLAMI instead of a minimal Ubuntu image? If so I'll change the AMI ids.

Reply:
There is no such requirement -- whatever works and is easy to implement will be good for this version. In a future update we can pick a suggested optimal AMI for production deployment. The DLAMIs have a lot of extra things so a lighter AMI targeted for production deployment will be good

Reply:
Will add this item to #88 to track for next release